Update to 16.5.0 and 2.1.0

nohwnd opened the pull request:
Done by replacing in Templates and template_feed folder.

dnfclas commented:
[![CLA assistant check](https://cla.dotnetfoundation.org/pull/badge/not_signed)](https://cla.dotnetfoundation.org/dotnet/test-templates?pullRequest=79) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/test-templates?pullRequest=79) before we can accept your contribution.<br/><br/>:x: nohwnd [sign now](https://cla.dotnetfoundation.org/dotnet/test-templates?pullRequest=79)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/test-templates?pullRequest=79) it.</sub>

nohwnd commented:
@singhsarab could you review this please? I bumped the version of testfx to 2.1.0 and test platform to 16.5.0. Added the .nupkg for mstest 2.1.0 and also deleted the UWP and EdgeDriver templates because you said we no longer own them and they also mentioned the packages I am updating.

singhsarab commented:
Edge driver is owned by the Redmond team, please check with @shyamnamboodiripad
UWP can be deleted.
But I would still encourage you to have separate PRs.

nohwnd commented:
@singhsarab Reverted the changes, but running the tests locally against 16.4.0 gives: 

```
AcceptanceTestBase.Execute: Stopped dotnet.exe. Exit code = 0


  X TemplateTest (template_feed\Microsoft.DotNet.Test.ProjectTemplates.2.2\content\MSTest-CSharp) [3s 358ms]
  Error Message:
   Assert.IsTrue failed. The Test summary does not match.
Expected summary: Total tests: 1. Passed: 1. Failed: 0. Skipped: 0
Test Output: C:\Program Files\dotnet\sdk\3.1.200-preview-014850\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.DefaultItems.targets(257,5): warning NETSDK1023: A PackageReference for 'Microsoft.NET.Test.Sdk' was included in your project. This package is implicitly referenced by the .NET SDK and you do not typically need to reference it from your project. For more information, see https://aka.ms/sdkimplicitrefs [C:\Projects\test-templates\artifacts\bin\Microsoft.TestTemplates.AcceptanceTests\Debug\netcoreapp2.1\template_feed\Microsoft.DotNet.Test.ProjectTemplates.2.2\content\MSTest-CSharp\Company.TestProject1.csproj] Test run for C:\Projects\test-templates\artifacts\bin\Company.TestProject1\Debug\netcoreapp2.2\Company.TestProject1.dll(.NETCoreApp,Version=v2.2) Microsoft (R) Test Execution Command Line Tool Version 16.3.0 Copyright (c) Microsoft Corporation. All rights reserved. Starting test execution, please wait... A total of 1 test files matched the specified pattern. Test Run Successful. Total tests: 1 Passed: 1 Total time: 0,9580 Seconds
Standard Error:
Arguments: template_feed\Microsoft.DotNet.Test.ProjectTemplates.2.2\content\MSTest-CSharp

  Stack Trace:
     at Microsoft.TestTemplates.AcceptanceTests.AcceptanceTestBase.ValidateSummaryStatus(Int32 passedTestsCount, Int32 failedTestsCount, Int32 skippedTestsCount) in C:\Projects\test-templates\test\Microsoft.TestTemplates.AcceptanceTests\AcceptanceTestBase.cs:line 63
   at Microsoft.TestTemplates.AcceptanceTests.DotnetCoreTemplateTests.TemplateTest(String path) in C:\Projects\test-templates\test\Microsoft.TestTemplates.AcceptanceTests\DotnetCoreTemplateTests.cs:line 31
```
Should I remove the references to test.sdk?

Also the outputs don't match: 
```
Assert.IsTrue failed. The Test summary does not match.
Expected summary: Total tests: 1. Passed: 1. Failed: 0. Skipped: 0
Total tests: 1 Passed: 1 Total time: 1,1788 Seconds
```
Is the output omitting failed and skipped if there are none?

Cosifne commented:
> 
> 
> @singhsarab could you review this please? I bumped the version of testfx to 2.1.0 and test platform to 16.5.0. Added the .nupkg for mstest 2.1.0 and also deleted the UWP and EdgeDriver templates because you said we no longer own them and they also mentioned the packages I am updating.
> 
> The build fails because we did not release 16.5.0 yet.

I believe the UWP templates could be removed but not sure why we want to remove the Edge Driver templates? Does the edge team requests to own that?

shyamnamboodiripad commented:
@Cosifne @nohwnd the Edge Driver templates need to be preserved.

nohwnd commented on test/Microsoft.TestTemplates.AcceptanceTests/DotnetCoreTemplateTests.cs:
```diff
@@ -13,12 +13,26 @@ public class DotnetCoreTemplateTests : AcceptanceTestBase
         /// <summary>
         /// The type of the test template, combination of the test framework and language
         /// </summary>
-        private static string[] templateTypes = { "MSTest-CSharp", "MSTest-FSharp", "MSTest-VisualBasic", "XUnit-CSharp", "XUnit-FSharp", "XUnit-VisualBasic" };
```
Formatting change, because it is easier to comment out the items from the arrays when you are debugging.

nohwnd commented on Templates/CSharp/EdgeDriverTemplate/DotNetCore/EdgeDriverTemplateCore/ProjectTemplates/CSharp/Test/EdgeDriverTemplate/EdgeDriverSample.csproj:
```diff
@@ -6,7 +6,7 @@
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="16.0.1" />
+    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="16.5.0-preview-20200203-01" />
```
This is a preview version. Let's merge it with that so I can delete the unneeded templates and still see the build pass. Then when 16.5.0 is released I will move to that version.

peterwald commented on template_feed/Microsoft.DotNet.Test.ProjectTemplates.2.0/content/MSTest-CSharp/Company.TestProject1.csproj:
```diff
@@ -7,12 +7,13 @@
 
     <IsPackable Condition="'$(EnablePack)' == 'true'">true</IsPackable>
     <IsPackable Condition="'$(EnablePack)' != 'true'">false</IsPackable>
+    <IsTestProject>true</IsTestProject>
```
Is this needed?

nohwnd replied:
nope removed everywhere.

peterwald commented on template_feed/Microsoft.DotNet.Test.ProjectTemplates.2.0/content/MSTest-FSharp/Company.TestProject1.fsproj:
```diff
@@ -8,6 +8,7 @@
     <IsPackable Condition="'$(EnablePack)' == 'true'">true</IsPackable>
     <IsPackable Condition="'$(EnablePack)' != 'true'">false</IsPackable>
     <GenerateProgramFile>false</GenerateProgramFile>
+    <IsTestProject>true</IsTestProject>
```
Also here....
